### PR TITLE
[DM-27146] Add a thorough README.md

### PR DESCRIPTION
**Pull request by cbanek**
This will help people at the bootcamp.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,115 @@
 # lsp-deploy
-A place to store the logic and configuration needed to spin up an entire LSP.
+
+This is the repository to store helper scripts and per-environment
+deployment configuration for the services of the Rubin Science
+Platform.
+
+# Background
```
Comment (jonathansick) [inline README.md:7]: I recommend h2-level headers for subsections.

Comment (cbanek) [reply README.md:7]: Sounds good to me!

File: README.md [hunk 2]
```diff
@@ -1,2 +1,115 @@
...
+layer here, and link to further documentation if you are interested.
+
+The Rubin Science Platform runs on Kubernetes (https://kubernetes.io/).
+Kubernetes provides a way to coordinate running services on multiple
+nodes.  Kubernetes runs a set of docker containers (https://docker.com/),
+and sets up the networking, storage, and configuration of those containers.
+
+Each different service in the Rubin Science Platform is a bit of code that
+eventually gets built and published into a docker container.  The git
+repositories hosting each service generally build their own docker containers
```
Comment (frossie) [inline README.md:20]: That sentence reads awkward (i think of something different that a git repo for "host a service")- how about something like "Git repositories for individual services typically have build pipelines resulting in new docker container builds when code changes are merged".

File: README.md [hunk 3]
```diff
@@ -1,2 +1,115 @@
...
+The configuration for a helm chart is called a values file, and is a simple
+YAML document that contains inputs for the templating of the chart.  Each
+environment has its own YAML document containing the required configuration
+for that chart.
+
+But Helm doesn't keep track of the application once it is deployed.  That is,
+it won't notice when the configuration changes and apply those changes.
+ArgoCD (https://argoproj.github.io/argo-cd/) fills this need.  ArgoCD watches
+this repository for new git commits and will keep track of those changes,
+either applying them automatically (sync'ing them), or waiting for an operator
```
Comment (frossie) [inline README.md:37]: ("syncing" them)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,115 @@
...
+creates all those services in argocd.  The values files in this directory
+contain the service manifest and other top level configuration.
+
+## How to use this repository
+
+Okay now that we understand the tools used, and the various conventions
+("The forms MUST be obeyed..."), we can get into how to use this repository
+to manage environments and services.
+
+There are already a number of different environments, such as int and
```
Comment (frossie) [inline README.md:80]: There are already a number of different environments reflecting our many different deployments, such as ...

**Review by frossie — APPROVED** [review]
Minor comments, looks great

File: README.md (hunk 2 above)
Comment (cbanek) [reply README.md:20]: Sounds good!

**Pull request merged**